src/action/shop.js

```diff
@@ -0,0 +1,56 @@
+import { SET_NEW_LIST_PRODUCT, SET_NEW_SEARCH_PARAMS } from "../constants";
+import { GET_START_PAGE } from "../constants";
+import apiFetch from "../utils/apiFetch";
+import { isEmpty } from "../utils/isEmpty";
+import { stringifySearchParams } from "../utils/stringifySearchParams";
+
+export const startPage = () => {
+  return async (dispatch, getState) => {
+    const { shop } = getState();
+    const { params } = shop;
+    const res = await apiFetch(`/products?${stringifySearchParams(params)}`);
+    const data = await res.json();
+    dispatch({
+      type: GET_START_PAGE,
+      payload: { [stringifySearchParams(params)]: data }
+    });
+  };
+};
```

Comment:
you really need `startPage` action ?

you just can call `changeParams` action with default params

and you can rename `changeParams` to `callListProductWithParams`